cdap-validation/bin/testing.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python
+
+import test_helpers as helpers
+import json
+import ambari
+import cm
+
+##### TEST FRAMEWORK FUNCTIONS #####
+
+input_json = "input.json"
+
+# calls all other functions
+def testing(base_vars):
+    print 'start testing'
+
+    # prep input variables
+    cluster_vars = {}
+    input = read_input(input_json)
+    cluster_vars['host'] = input['params']['uri']
+    cluster_vars['username'],cluster_vars['password']= input['params']['userpass'].split(':')
+    cluster_vars['cluster'] = input['params']['cluster_name']
+    cluster_vars['modules'] = input['params']['modules']
+
+    # check for verbosity not set
+    if not input['params']['verbose']:
+        cluster_vars['verbose'] = 'default'
+    else:
+        cluster_vars['verbose'] = input['params']['verbose']
+
+    # find_modules
+    # hold on for now
+
+    # detect_installed_services
+    # hold on for now
+
+    # prep the url
+    find_api='/api'
+    if cluster_vars['host'].find(find_api) == -1:
+        cluster_vars['host'] += 'api/'
+    print 'uri=%s' % (cluster_vars['host'])
+
+    # validate_connection
+    print 'Validating connection'
+    api_tests = [{'cm': base_vars['cm']['api_test']},{'ambari': base_vars['ambari']['api_test']}]
+    version_test = validate_connection(base_vars, cluster_vars, api_tests)
```

Comment:
if you are passing ``base_vars`` already, you shouldn't need to also pass ``api_tests`` which are derived from ``base_vars`` in the line above.  It looks like in general ``base_vars`` is passed around alot (renamed as ``base_info``)... seems like it should just accessible to all via some other mechanism (class variables, a config class that gets included, etc... whatever is typical in python)